unitTests/ui/SimpleFilterTest.ts

```diff
@@ -224,5 +228,90 @@ export function SimpleFilterTest() {
+
+      function clickOnClearButton() {
+        $$(simpleFilterContainer.find('.coveo-simplefilter-eraser')).trigger('click');
+      }
+
+      function selectSomeValues() {
+        simpleFilterWithEraser.cmp.selectValue('foo');
+        simpleFilterWithEraser.cmp.selectValue('bar');
+      }
+
+      function getClearButton() {
+        return $$(simpleFilterContainer.find('.coveo-simplefilter-eraser'));
+      }
+
+      beforeEach(() => {
+        simpleFilterWithEraser = Mock.optionsComponentSetup<SimpleFilter, ISimpleFilterOptions>(SimpleFilter, <ISimpleFilterOptions>{
+          ...aSimpleFilter.cmp.options,
+          enableClearButton: true
+        });
+        simpleFilterContainer = $$(simpleFilterWithEraser.cmp.element);
+        simpleFilterWithEraser.cmp.toggleContainer();
+      });
+
+      it('should create a clear button if enableClearButton is true', () => {
+        expect(getClearButton()).not.toBeFalsy();
+      });
+
+      it('should display the clear button if there are selected values', () => {
+        selectSomeValues();
+
+        expect(getClearButton().isVisible()).toBe(true);
+      });
+
+      it('should hide the clear button after deselecting all values', () => {
+        selectSomeValues();
+        simpleFilterWithEraser.cmp.deselectValue('foo');
+        simpleFilterWithEraser.cmp.deselectValue('bar');
+
+        expect(getClearButton().isVisible()).toBe(false);
+      });
+
+      it('should deselect all values after clicking on the clear button', () => {
+        selectSomeValues();
+        clickOnClearButton();
+
+        expect(simpleFilterWithEraser.cmp.getSelectedValues().length).toBe(0);
+      });
+
+      it('should execute only one query after clicking on the clear button', () => {
+        selectSomeValues();
+        (simpleFilterWithEraser.cmp.queryController.executeQuery as jasmine.Spy).calls.reset();
+        clickOnClearButton();
+
+        expect(simpleFilterWithEraser.cmp.queryController.executeQuery).toHaveBeenCalledTimes(1);
+      });
+
+      it('should log search event when a clear button is clicked', () => {
+        selectSomeValues();
+        (simpleFilterWithEraser.env.usageAnalytics.logSearchEvent as jasmine.Spy).calls.reset();
+
+        clickOnClearButton();
+
+        expect(simpleFilterWithEraser.env.usageAnalytics.logSearchEvent).toHaveBeenCalledWith(
+          jasmine.objectContaining({
+            name: 'clearAll',
+            type: 'simpleFilter'
+          }),
+          jasmine.objectContaining({
+            simpleFilterTitle: simpleFilterWithEraser.cmp.options.title
+          })
+        );
+      });
+
+      it('should log only one search event after clicking on clear button', () => {
+        selectSomeValues();
+        (simpleFilterWithEraser.env.usageAnalytics.logSearchEvent as jasmine.Spy).calls.reset();
+        clickOnClearButton();
+
+        expect(simpleFilterWithEraser.env.usageAnalytics.logSearchEvent).toHaveBeenCalledTimes(1);
```

Comment:
I generally like to keep one expectation per test too. In this case though, I think it's okay to combine this line with the previous test. My thinking is that when the line is removed, only one test fails, but having two tests is fine too.